### PR TITLE
Fixes DateRange by using suppliers as they are often used in a static context

### PR DESCRIPTION
**Pull request by fhaScireum**
- FIxes: OX-6320

### CODE REVIEW

File: src/main/java/sirius/db/mixing/DateRange.java [hunk 1]
```diff
@@ -276,17 +294,19 @@ public LocalDateTime getUntil() {
      */
     public <C extends Constraint> void applyTo(String field, boolean useLocalDate, Query<?, ?, C> query) {
         List<C> constraints = new ArrayList<>(2);
-        if (from != null) {
+        if (fromSupplier != null) {
```
Comment (sabieber) [inline src/main/java/sirius/db/mixing/DateRange.java:297]: hier sollten auch die getFrom und getUntil aufgerufen werden und im if geprüft werden ob die resultierenden localdatetimes null sind. sonst ändert sich das verhalten durch den PR

**Review by sabieber — CHANGES_REQUESTED** [review]

File: src/main/java/sirius/db/mixing/DateRange.java [hunk 2]
```diff
@@ -248,7 +266,7 @@ public String getKey() {
      * @return the start of the date range
      */
     public LocalDateTime getFrom() {
-        return from;
+        return fromSupplier != null ? fromSupplier.get() : null;
```
Comment (idlira) [inline src/main/java/sirius/db/mixing/DateRange.java:269]: What situation could actually bring such a supplier to be null?

File: src/main/java/sirius/db/mixing/DateRange.java (hunk 1 above)
Comment (idlira) [reply src/main/java/sirius/db/mixing/DateRange.java:297]: What could make such a supplier return null in first place?

**Review by idlira — CHANGES_REQUESTED** [review]
...

File: src/main/java/sirius/db/mixing/DateRange.java (hunk 2 above)
Comment (fhaScireum) [reply src/main/java/sirius/db/mixing/DateRange.java:269]: If the constructor would be called without a from supplier :)

**Comment by fhaScireum** [comment]
> A small improvement to the class could be a `.withReferenceDate` which is used if defined instead of LocalDateTime.now(), allowing a caller to actually freeze dates if really desired.

In that case the user could call the constructor with the until or from supplier always returning a specific date. I think calling a quasi constructors with a descriptive method name and overruling one of the two essential dates is quite messy.

**Review by idlira — APPROVED** [review]

File: src/main/java/sirius/db/mixing/DateRange.java (hunk 1 above)
Comment (sabieber) [reply src/main/java/sirius/db/mixing/DateRange.java]: Dont know, but better safe than sorry. The null checks are already in place

**Review by sabieber — APPROVED** [review]

File: src/main/java/sirius/db/mixing/DateRange.java [hunk 3]
```diff
@@ -55,7 +59,10 @@ public DateRange(String key, String name, @Nullable LocalDateTime from, @Nullabl
      * @return a date range for the given interval
      */
     public static DateRange lastFiveMinutes() {
```
Comment (andyHa) [inline src/main/java/sirius/db/mixing/DateRange.java:61]: shouldn't this be constants then? (all factory methods)

Comment (andyHa) [reply src/main/java/sirius/db/mixing/DateRange.java:61]: maybe we should add constants and make the methods deprecated as a first step

File: src/main/java/sirius/db/mixing/DateRange.java [hunk 4]
```diff
@@ -29,24 +30,27 @@
...
-     * @param key   the unique name of the ranged used as filter value
-     * @param name  the translated name shown to the user
-     * @param from  the lower limit (including) of the range
-     * @param until the upper limit (including) of the range
+     * @param key           the unique name of the ranged used as filter value
+     * @param nameSupplier  the translated name shown to the user
+     * @param fromSupplier  the lower limit (including) of the range
+     * @param untilSupplier the upper limit (including) of the range
      */
-    public DateRange(String key, String name, @Nullable LocalDateTime from, @Nullable LocalDateTime until) {
```
Comment (andyHa) [inline src/main/java/sirius/db/mixing/DateRange.java:45]: add a legacy constructor which invokes a new one and mark this deprecated? this way we prevent a breaking change...

**Review by andyHa — CHANGES_REQUESTED** [review]
...

File: src/main/java/sirius/db/mixing/DateRange.java [hunk 5]
```diff
@@ -29,138 +30,206 @@
...
 
     private final String key;
-    private final String name;
-    private final LocalDateTime from;
-    private final LocalDateTime until;
+    private final Supplier<String> nameSupplier;
+    private final Supplier<LocalDateTime> fromSupplier;
+    private final Supplier<LocalDateTime> untilSupplier;
+
+    public static final DateRange LAST_FIVE_MINUTES = lastFiveMinutes();
```
Comment (andyHa) [inline src/main/java/sirius/db/mixing/DateRange.java:37]: javadoc + don't call depracated code here...

**Review by idlira — APPROVED** [review]

**Review by andyHa — CHANGES_REQUESTED** [review]
...

**Pull request closed without merging**